builtin/providers/aws/structure.go

```diff
@@ -207,3 +207,21 @@ func expandStringList(configured []interface{}) []string {
 	}
 	return vs
 }
+
+// Flattens an array of DBSecurityGroups into a []string
+func flattenEC2SecurityGroupStatuses(list []rds.EC2SecurityGroup) []string {
+	result := make([]string, 0, len(list))
+	for _, i := range list {
+		result = append(result, *i.Status)
+	}
+	return result
+}
+
+// Flattens an array of IPRanges into a []string
+func flattenIPRangeStatuses(list []rds.IPRange) []string {
+	result := make([]string, 0, len(list))
+	for _, i := range list {
+		result = append(result, *i.Status)
+	}
+	return result
+}
```

Comment:
I needed to flatten the Security Group and IPRange statuses down, so I added these here. 
I thought later that they were simple enough that maybe I should have just added them in the func that calls them?

Is there a more idiomatic way to do this in a single function, since the structure is the same and the field `Status` is the same?